# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: uriloader/preload/PreloadHashKey.cpp
# Commit: c11cffaded9c
# Full Hash: c11cffaded9cfed53b04e5c7875278e9a9e55396
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r 0527d227c6b6 -r c11cffaded9c uriloader/preload/PreloadHashKey.cpp
--- a/uriloader/preload/PreloadHashKey.cpp	Fri Feb 18 12:45:52 2022 +0000
+++ b/uriloader/preload/PreloadHashKey.cpp	Fri Feb 18 12:45:53 2022 +0000
@@ -75,8 +75,8 @@
 }
 
 // static
-PreloadHashKey PreloadHashKey::CreateAsScript(nsIURI* aURI, CORSMode aCORSMode,
-                                              dom::ScriptKind aScriptKind) {
+PreloadHashKey PreloadHashKey::CreateAsScript(
+    nsIURI* aURI, CORSMode aCORSMode, JS::loader::ScriptKind aScriptKind) {
   PreloadHashKey key(aURI, ResourceType::SCRIPT);
   key.mCORSMode = aCORSMode;
 
@@ -89,9 +89,9 @@
 PreloadHashKey PreloadHashKey::CreateAsScript(nsIURI* aURI,
                                               const nsAString& aCrossOrigin,
                                               const nsAString& aType) {
-  dom::ScriptKind scriptKind = dom::ScriptKind::eClassic;
+  JS::loader::ScriptKind scriptKind = JS::loader::ScriptKind::eClassic;
   if (aType.LowerCaseEqualsASCII("module")) {
-    scriptKind = dom::ScriptKind::eModule;
+    scriptKind = JS::loader::ScriptKind::eModule;
   }
   CORSMode crossOrigin = dom::Element::StringToCORSMode(aCrossOrigin);
   return CreateAsScript(aURI, crossOrigin, scriptKind);